knowledge-base/adr/0033-send-email.md

```diff
@@ -0,0 +1,118 @@
+# Title
+Send email by using destination service on BTP
+
+## Status
+
+approved
+
+## Context
+
+BTP destination service supports a type `MAIL`, so application on the cloud can send mails with a given mail server.
+See the type of the `Destination` object [in our code](https://github.com/SAP/cloud-sdk-js/blob/main/packages/connectivity/src/scp-cf/destination/destination-service-types.ts#L95).
+
+However, the type `MAIL` is not supported by the SDK for the time being.
+
+The scope of this documentation covers the following proxy type:
+- `Internet`
+- `OnPremise`
+
+## Decision
+### Package management
+We assume, the mail function will not be used by everybody.
+Therefore, we'll create a new package called `mail-client`, which has the following dependencies:
+- `@sap-cloud-sdk/connectivity`
+- `nodemailer`: 473 KB
+- `socks`: 320 KB
+
+### API
+#### Public API
+As no Node native mail interface was found, we can bind our implementation to a common lib like `nodemailer` for the initial version.
+```ts
+export function sendMail<T extends MailOptions>(
```

Comment:
@marikaner @FrankEssenberger 

In SDK, we commonly use generic types because of better extensibility.
See example [here](https://www.typescriptlang.org/play?target=5&ts=4.1.5#code/JYOwLgpgTgZghgYwgAgLIQM4bgcwgbwFgAoZM5MAewC5kMwpQcAaE85BS8CcAflvqMQOEgF8SJUJFiIUqXMATosuFBAAekEABMMaTNjxFS5ALYKEAhkwDcYicRgBXEAjDAudHtuWGIAQQAKUwwcWl9VAEpwg1VjdigIMCcoEGQQnDticWISZ1d3TwxvCLwAIQAeABVkDS1dfRU8AD5g0Noq6OQq+PJE5NT00KyckgB6MeQ4bW1gQpA4ABtkAAcoShXod0wpxOQQSjApxcXKAHcIbXHJ-pS0sABPTeRtSh2Do5dtaHo4HWQAAbmHCKAFTPR-KYzOYeBbLNYbLYPEjFHSlAKBfBUWgAchxzHSFlxONEkQAdMDFFlUT5YuVMdjkHiCZTLEySeTWTYgA)

Here is the copy, as the link might expire.
```
interface Message{
    to: string,
    content?: string
}

interface MagicMessage extends Message{
    magic: string;
}

function sendMessageA(msg: Message): Message{
    return msg;
}

function sendMessageB<T extends Message>(msg: T): T{
    return msg;
}

// additional properties are not allowed
// return type does not understand `magic` as an additional property
sendMessageA({to: '', magic: ''}).magic;
sendMessageB({to: '', magic: ''}).magic;
```